src/test/java/edu/hm/hafner/analysis/ast/AbstractAstTest.java

```diff
@@ -0,0 +1,223 @@
+    static final String LINE77_METHOD = "METHOD_DEF MODIFIERS LITERAL_PRIVATE TYPE LITERAL_VOID IDENT LPAREN PARAMETERS RPAREN SLIST ";
+    static final String LINE78_METHOD_CALL =  "EXPR METHOD_CALL DOT DOT IDENT IDENT IDENT ELIST EXPR STRING_LITERAL RPAREN SEMI ";
+    static final String LINE79_NESTED_METHOD_RCURLY =  "RCURLY ";
+    static final String LINE80_NESTED_CLASS_RCURLY =  "RCURLY ";
+    static final String NESTED = LINE76_NESTED + LINE77_METHOD + LINE78_METHOD_CALL + LINE79_NESTED_METHOD_RCURLY + LINE80_NESTED_CLASS_RCURLY;
+
+    static final String WHOLE_METHOD = LINE67_METHOD + LINE68_VAR + LINE69_VAR
+            + LINE70_VAR + LINE71_VAR + LINE73_VAR + LINE75_VAR + LINE76_IF + LINE77_ASSIGN + LINE78_RCURLY
+            + LINE79_ELSE
+            + LINE80_ASSIGN + LINE81_RCURLY + LINE82_FOR + LINE83_IF + LINE84_IF + LINE85_ASSIGN + LINE86_RCURLY
+            + LINE87_ELSE + LINE88_ASSIGN + LINE89_RCURLY
+            + LINE90_CALL + LINE91_CALL + LINE92_IF + LINE93_INCR + LINE94_CALL + LINE95_CALL + LINE96_RCURLY
+            + LINE97_RCURLY + LINE98_ELSE + LINE99_BREAK + LINE100_RCURLY
+            + LINE101_RCURLY + LINE103_RETURN + LINE104_RCURLY;
+    static final String WHOLE_CLASS = LINE14_CLASS + LINE16_FIELD + LINE18_FIELD
+            + LINE25_CTOR + LINE26_SUPER + LINE27_RCURLY + LINE29_METHOD
+            + LINE30_RETURN + LINE31_RCURLY + LINE33_METHOD + LINE34_RETURN + LINE35_RCURLY
+            + WHOLE_METHOD + LINE105_RCURLY;
+    static final String WHOLE_FILE = LINE1_PACKAGE + LINE3_IMPORT + LINE4_IMPORT + LINE6_IMPORT + LINE14_CLASS + LINE16_FIELD + LINE18_FIELD
+            + LINE25_CTOR + LINE26_SUPER + LINE27_RCURLY + LINE29_METHOD
+            + LINE30_RETURN + LINE31_RCURLY + LINE33_METHOD + LINE34_RETURN + LINE35_RCURLY
+            + WHOLE_METHOD + NESTED + LINE105_RCURLY;
+
+    Ast createAst(final int lineNumber) {
+        String fileName = createJavaSourceTemporaryFile("elements.ast-test");
+
+        return createAst(fileName, lineNumber);
+    }
+
+    /**
+     * Creates the AST under test for the specified file at the given line number.
+     *
+     * @param fileName
+     *         the file to scan
+     * @param lineNumber
+     *         the first interesting line number
+     *
+     * @return the created {@link Ast}
+     */
+    abstract Ast createAst(String fileName, int lineNumber);
+
+    void assertThatAstIs(final Ast ast, final String expectedResult) {
+        assertThat(ast.chosenAreaAsString(' ')).isEqualTo(expectedResult);
+    }
+
+    String createJavaSourceTemporaryFile(final String fileName) {
+        File warnings = createCopyInTemp(fileName);
+        return warnings.getAbsolutePath();
+    }
+
+    private File createCopyInTemp(final String fileName) {
+        try {
+            File sourceCode = File.createTempFile("ast", ".java");
+            sourceCode.deleteOnExit();
+
+            InputStream stream = AbstractAstTest.class.getResourceAsStream(fileName);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=5019963)